src/EditorFeatures/CSharpTest/MakeFieldReadonly/MakeFieldReadonlyTests.cs

```diff
@@ -0,0 +1,616 @@
+        public async Task FieldIsPublic()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        public int [|_foo|];
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldIsInternal()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        internal int [|_foo|];
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldIsEvent()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private event System.EventHandler [|Foo|];
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldIsReadonly()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private readonly int [|_foo|];
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldNotAssigned()
+        {
+            await TestInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private int [|_foo|];
+    }
+}",
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private readonly int _foo;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldNotAssigned_Struct()
+        {
+            await TestInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    struct MyStruct
+    {
+        private int [|_foo|];
+    }
+}",
+@"namespace ConsoleApplication1
+{
+    struct MyStruct
+    {
+        private readonly int _foo;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task FieldAssignedInline()
+        {
+            await TestInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private int [|_foo|] = 0;
+    }
+}",
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private readonly int _foo = 0;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeFieldReadonly)]
+        public async Task MultipleFieldsAssignedInline_AllCanBeReadonly()
+        {
+            await TestInRegularAndScriptAsync(
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private int [|_foo|] = 0, _bar = 0;
+    }
+}",
+@"namespace ConsoleApplication1
+{
+    class MyClass
+    {
+        private int _bar = 0;
+        private readonly int _foo = 0;
```

Comment:
@CyrusNajmabadi I'm not worried. This is a relatively rare case. A follow-up issue could be added to make the Fix All handle the case differently where possible.

❗️ @Hosch250 The part I definitely don't like here is the reordering of the initializers. When splitting multiple initializers, the order should not change. This is required in order to preserve the semantics of the original statement during the change. Make sure to test all three cases:

```
private int _changed, _extra;
private int _extra, _changed;
private int _extra1, _changed, _extra2;
```

In the latter case the result should be this:

```csharp
private int _extra1;
private readonly int _changed;
private int _extra2;
```